src/infer_request.cc

```diff
@@ -881,8 +881,14 @@ InferenceRequest::Normalize()
                 "'");
       }
 
-      input.MutableShape()->assign(
-          input.OriginalShape().begin() + 1, input.OriginalShape().end());
+      // When skipping the first dimension, make sure there is a second
+      // dimension
+      if (input.OriginalShape().size() > 1) {
```

Comment:
Do you actually observe segfault without this change? I would imagine the iterator (`xxx.begin() + 1`)will be out of range only if the size is 0. If size is 1, then `begin() + 1` should equivalent to `end()` and `assign()` won't do anything?

Reply:
Perhaps changing the check in line 866 to `if (input.OriginalShape().size() < 2) {`? At this point we should assume the input has at least two dimensions (input config shape which is > 0, + batch dim), although the error message needs to be changed to reflect that

Reply:
Sounds good, updated.
What do you think of the updated error message?